# Summer2021_mbov. 

## Authors
Noah A. Legall, Liliana Salvador (advisor)

## Purpose 
This github is tied to the manuscript "Comparative genomic analysis of worldwide Mycobacterium bovis isolates reveal geographic and host associated evolution". Programs that were used to generate data for the analysis can be found in the 'scripts' directory. Other necessary files are located in the 'dependencies' file 

## Abstract 
Mycobacterium bovis, a bacterial zoonotic pathogen responsible for the economically and agriculturally important livestock disease bovine tuberculosis (bTB), has shown characteristics of a generalist pathogen, infecting worldwide a broad range of mammalian species. These characteristics have led to bidirectional transmission events between livestock and wildlife species and to the formation of wildlife reservoirs, impacting the ability of bTB control measures. Next Generation Sequencing (NGS) has transformed our ability to understand disease transmission events by tracking variant sites. However, factors for host adaptation following spillover, alongside the role of other pathogen genomic processes in the M. bovis transmission process are understudied problems. In this study, we analysed three rich publicly available datasets (700 isolates) of  M. bovis samples collected from both livestock and wildlife (11 host-species represented) from the United Kingdom, United States of America, and New Zealand to investigate if specific M. bovis genomic signatures (such as gene flow, homologous recombination, or sites of positive selection) are related to the emergence and evolution of M. bovis at the host-species, geographical, and/or sub-population levels. Regions that were predicted to be impacted by homologous recombination in the M. bovis genome were found to affect multiple pe/ppe genes, which in previous studies have been indicated as having functions for pathogenicity. More stuff to come
