---
title: "Mbovis_HR_analysis"
author: "Noah Legall"
date: "6/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Visualizing Regions Impacted by Homologous Recombination 
load in the heatmap data

```{r}
library(reshape)
library(ggplot2)
library(gt)
library(dplyr)
library(ggtree)
library(phytools)
library(patchwork)
library(tidyr)
```

```{r}
setwd("../dependencies/")
recomb_heat = read.csv("recombination_heatmap.csv",header = TRUE)

pre_regions <- as.data.frame(colnames(recomb_heat))
pre_regions$regions <- gsub("X","",pre_regions$`colnames(recomb_heat)`)
pre_regions <- subset(pre_regions,pre_regions$regions != "label")
regions <- pre_regions %>% separate(regions,c("start","end"))
regions$start <- as.numeric(regions$start)
regions$end <- as.numeric(regions$end)
regions$coords <- floor((regions$end + regions$start)/2)
regions <- regions[order(regions$coords),]
colnames(recomb_heat) <- c("label",regions$coords)

regions %>% ggplot(aes(x = coords)) + 
  geom_point(aes(y = 0),color = "red", size = 3) + 
  theme(axis.title.y=element_blank(),
           axis.text.y=element_blank(), 
           axis.ticks.y=element_blank(),
        panel.background = element_blank()) +
  xlim(0, 4.4e+06)


recomb.melted = melt(recomb_heat, id="label")
#head(recomb.melted)
ggplot(recomb.melted, aes(variable,label,fill=factor(value))) + 
  geom_tile() + 
  scale_fill_manual(label=c("absent","present"),values=c("white","red")) +
  theme(axis.text.x=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks =element_blank(),
        ) +
  ggtitle("Heatmap of genomic regions impacted by homologous recombination") +
  xlab("genomic region") + 
  ylab("sample") +
  labs(fill = "evidence of HR")
  
```
```{r}
pre_regions <- as.data.frame(colnames(recomb_heat))
pre_regions$regions <- gsub("X","",pre_regions$`colnames(recomb_heat)`)
pre_regions <- subset(pre_regions,pre_regions$regions != "label")
regions <- pre_regions %>% separate(regions,c("start","end"))
regions$start <- as.numeric(regions$start)
regions$end <- as.numeric(regions$end)
regions <- regions[order(regions$start),]
regions$coords <- floor((regions$end + regions$start)/2)
regions$color <- "red"

regions %>% ggplot(aes(x = coords)) + 
  geom_point(aes(y = 0),color = "red", size = 3) + 
  theme(axis.title.y=element_blank(),
           axis.text.y=element_blank(), 
           axis.ticks.y=element_blank(),
        panel.background = element_blank()) +
  xlim(0, 4.4e+06)

```


```{r}
recomb_genes <- read.table("recomb_gene.tbl",col.names = paste0("V",seq_len(10)), fill=TRUE)
head(recomb_genes) 
region <- gsub(".bed","",gsub("_","-",recomb_genes$V1))
gene_name <- gsub(";gbkey","",gsub("Name=","",stringr::str_extract(string = recomb_genes$V10,pattern = "Name=.*;gbkey")))
hr_impact.tbl <- data.frame(region,gene_name)
hr_impact.tbl %>% gt( ) %>% tab_header(title = "Mycobacterium bovis genes impacted by homologous recombination") 
#tbl_out %>% gtsave("hr_table.html",inline_css = TRUE)
```

```{r}
setwd("../dependencies/")
hr_stats <- read.csv("recombination_stats.csv")
hr_stats %>% ggplot(aes(x=Import.Length,)) + 
  geom_histogram(binwidth = 10,fill = "steelblue") +
  theme_minimal() +
  xlab("size of the homologous recombination region")

hr_stats %>% ggplot(aes(x=No..SNPs,)) + 
  geom_histogram(binwidth = 1,fill = "steelblue") +
  theme_minimal() +
  xlab("No. of SNPs within a homologous recombination region")

import_num <- read.csv("isolate_import_num.csv",header = TRUE)
accession_meta <- read.csv("ProjectBovisReservoir_metadata.csv", header = TRUE, stringsAsFactors = FALSE)
import_num <- import_num %>% left_join(accession_meta,by = c("label"= "Sample"))

country_box <- ggplot(import_num, aes(x=Country, y=X0, fill=Country)) + 
  geom_boxplot() +
  labs(y="Number of homologous recombination regions") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                        colour = "gray80"), 
        panel.grid.minor = element_line(size = 0.25, linetype = 'solid',
                                        colour = "gray80")) 

species_box <- ggplot(subset(import_num,import_num$Species != "STOAT-NZ"), aes(x=Species, y=X0, fill=Species)) + 
  geom_boxplot() + 
  labs(y="Number of homologous recombination regions") +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_line(size = 0.5, linetype = 'solid',
                                        colour = "gray80"), 
        panel.grid.minor = element_line(size = 0.25, linetype = 'solid',
                                        colour = "gray80"))
country_box
```

```{r}
mbov_tree <- read.tree("../dependencies/mbov_align.final_tree.tre")
isolate_dat = read.csv("../dependencies/filtered_isolate_list.csv",header = TRUE)
isolate_melt <- melt(isolate_dat,id = "Sample")
isolate_melt$variable=as.character(levels(isolate_melt$variable))[isolate_melt$variable]

mbov_tree <- root(mbov_tree,"Reference")
mbov_tree <- drop.tip(mbov_tree,"Reference")
mbov <- ggtree(mbov_tree,layout = "circular",branch.length = 'none') 

tree_meta <- read.csv("../dependencies/filtered_isolate_list.csv", header = TRUE)

mbov %<+% tree_meta +
  geom_tippoint(aes(color = Species))

#taxa_order <- as.data.frame(mbov_tree$tip.label)
#taxa_order <- taxa_order %>% left_join(tree_meta,by = c("mbov_tree$tip.label"= "Sample"))
#nz_samps = subset(tree_meta,tree_meta$Country == "New Zealand")
#I used the taxa_order to find the clades to subset

#find the index of the isolates, and chose the max & min 
#viewClade(mbov,MRCA(mbov,"SRX5400248","SRX5400221"))
#viewClade(mbov,MRCA(mbov,"SRX2526938","SRX2527172"))
```

