---
title: "Mbovis_HR_analysis"
author: "Noah Legall"
start date: "6/21/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Libraries

```{r}
library(reshape)
library(ggplot2)
library(gt)
library(dplyr)
library(ggtree)
library(phytools)
library(patchwork)
library(tidyr)
library(fastbaps)
library(ape)
library(ggnewscale)
library(gridExtra)
```
## Homologous recombination heatmap

```{r}
setwd("../dependencies/")
recomb_heat = read.csv("recombination_heatmap.csv",header = TRUE)

#Processing of the gubbins heatmap data
pre_regions <- as.data.frame(colnames(recomb_heat))
pre_regions$regions <- gsub("X","",pre_regions$`colnames(recomb_heat)`)
pre_regions <- subset(pre_regions,pre_regions$regions != "label")
regions <- pre_regions %>% separate(regions,c("start","end"))
regions$start <- as.numeric(regions$start)
regions$end <- as.numeric(regions$end)
regions$coords <- floor((regions$end + regions$start)/2)
regions <- regions[order(regions$coords),]
colnames(recomb_heat) <- c("label",regions$coords)

#Regions that are duplicated will be ignored, and the columns will be ordered 
recomb_heat <- recomb_heat[,!duplicated(colnames(recomb_heat))]
recomb_heat <- recomb_heat[,order(as.numeric(names(recomb_heat)))]


#Create the heatmap using ggplot2
recomb.melted = melt(recomb_heat, id="label")
ggplot(recomb.melted, aes(variable,label,fill=factor(value))) + 
  geom_tile() + 
  scale_fill_manual(label=c("absent","present"),values=c("white","black")) +
  theme(text = element_text(size = 15),
        axis.text.x=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks =element_blank(),
        legend.key = element_rect(fill = "white", colour = "black")
        ) +
  xlab("genomic region") + 
  ylab("sample") +
  labs(fill = "evidence of HR")
  
```

## HR stats

```{r}
setwd("../dependencies/")

#Generate histograms to detail the characteristics of predicted HR regions from Gubbins.
recomb_heat_nolab <-recomb_heat[,-which(names(recomb_heat) %in% c("label"))]
recomb_heat_nolab$num_hr <- rowSums(recomb_heat_nolab)
num_genomes <- as.data.frame(colSums(recomb_heat_nolab))
colnames(num_genomes)[1] <- "num"


hr_stats <- read.csv("recombination_stats.csv")

import_gg <- hr_stats %>% ggplot(aes(x=Import.Length,)) + 
  geom_histogram(binwidth = 10,fill = "gray") +
  theme_minimal() +
  xlab("HR length") + labs(tag = "A")

snp_size_gg <- hr_stats %>% ggplot(aes(x=No..SNPs,)) + 
  geom_histogram(binwidth = 1,fill = "gray") +
  theme_minimal() +
  xlab("SNPs per HR region") + labs(tag = "B")

a <- recomb_heat_nolab %>% ggplot(aes(x=num_hr)) +
  geom_histogram(binwidth=1,fill = "gray") + 
  theme_minimal() +
  xlab("HR per isolate") + labs(tag = "C")

b <- num_genomes %>% filter(num < 700)  %>% ggplot(aes(x=num)) +
  geom_histogram(binwidth=10,fill = "gray") + 
  theme_minimal() +
  xlab("isolates with a specifc HR region") + labs(tag = "D")
  labs(tag = "B")


grid.arrange(import_gg,snp_size_gg,a,b,nrow=2)

```

## Population Structure

```{r}
#Utilize fastBAPS to infer clustering of isolates given the SNP data. 
mbov_snp <- "../dependencies/mbov_snp.fasta"
sparse.data <- import_fasta_sparse_nt(mbov_snp)
sparse.data <- optimise_prior(sparse.data, type = "optimise.symmetric")
baps.hc <- fast_baps(sparse.data)
best.partition <- best_baps_partition(sparse.data, baps.hc)

iqtree <- phytools::read.newick("../dependencies/mbov_align.contree")
plot.df <- data.frame(id = colnames(sparse.data$snp.matrix), fastbaps = best.partition, 
    stringsAsFactors = FALSE)

#Metadata file presented to isolate_dat
isolate_dat = read.csv("../dependencies/filtered_isolate_list.csv",header = TRUE)
isolate_dat <- isolate_dat %>% left_join(plot.df,by=c("Sample"="id"))

#Quick visualization
gg <- ggtree(iqtree)
pop_clust <- facet_plot(gg, panel = "fastbaps", data = plot.df, geom = geom_tile, aes(x = fastbaps), color = "blue")
pop_clust
```


## Phylogeny
```{r}
# Tree developed from SNP alignment of M. bovis alignment 
mbov_tree <- read.newick("../dependencies/mbov_align.contree")


#change labels 
old_bov <- c("BOVINE-Michigan","BOVINE-UK","BOVINE-NZ")
for(i in 1:length(old_bov)){
  isolate_dat$Species <- gsub(old_bov[i],"BOVINE",isolate_dat$Species)
}

old_cerv <- c("WTD-Michigan","ELK-Michigan","CERVINE-NZ")
for(i in 1:length(old_cerv)){
  isolate_dat$Species <- gsub(old_cerv[i],"CERVINE",isolate_dat$Species)
}

isolate_dat$Species <- gsub("-UK","",isolate_dat$Species)
isolate_dat$Species <- gsub("-NZ","",isolate_dat$Species)
isolate_dat$Species <- gsub("-US","",isolate_dat$Species)


mbov_tree <- root(mbov_tree,"LT708304.1")
mbov_tree <- drop.tip(mbov_tree,"LT708304.1")
mbov <- ggtree(mbov_tree) 

#Generate the data that will become rings that will surround the phylogeny
baps_heat <- as.data.frame(as.factor(isolate_dat$fastbaps))
rownames(baps_heat) <- isolate_dat$Sample

species_heat <- as.data.frame(as.factor(isolate_dat$Species))
rownames(species_heat) <- isolate_dat$Sample

country_heat <- as.data.frame(as.factor(isolate_dat$Country))
rownames(country_heat) <- isolate_dat$Sample


#color palettes
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
cb_col <- colorRampPalette(cbPalette)
species_col <- palette.colors(7)
country_col <- hcl.colors(3,"SunsetDark")



#ggtree to label nodes based on bootstrap support
mbov <- ggtree(mbov_tree, layout = "circular", branch.length = "none")
mbov_node_support <- mbov %<+% isolate_dat + 
  geom_point2(aes(color=col), show.legend = TRUE) + 
  geom_point2(aes(label=label, subset = isTip, color = "tip")) + 
  scale_color_manual(values = c("red","maroon","gray75","black"), name = "Bootstrap Support")
  
t1 <- gheatmap(mbov_node_support,baps_heat,width = 0.1, colnames = FALSE, legend_title = "Population Cluster") + scale_fill_manual(values = cb_col(8), name = "Population Cluster")
t2 <- t1 + new_scale_fill()
t2_1 <- gheatmap(t2,species_heat,offset = 5, width = 0.1, colnames = FALSE, legend_title = "Species") +  scale_fill_manual(values = species_col, name = "Species")
t3 <- t2_1 + new_scale_fill()
mbov_final <- gheatmap(t3,country_heat,offset = 10, width = 0.1, colnames = FALSE, legend_title = "Country") + 
  scale_fill_manual(values = country_col, name = "Country")

mbov_final
```

##Boxplots
```{r}
# Describe the distributions of HR regions per species.
import_num <- read.csv("../dependencies/isolate_import_num.csv",header = TRUE)
accession_meta <- read.csv("../dependencies/filtered_isolate_list.csv", header = TRUE, stringsAsFactors = FALSE)
import_num <- import_num %>% left_join(isolate_dat,by = c("label"= "Sample"))
import_num$X0 <- recomb_heat_nolab$num_hr

country_no_lookup <- import_num %>% group_by(Country) %>% summarise(n=n()) %>% mutate(country_ext = paste0(Country," (",n,")"))
import_num <- import_num %>% left_join(country_no_lookup,by = c("Country"))
country_box <- ggplot(import_num, aes(x=country_ext, y=X0, fill=Country)) + 
  geom_boxplot(color="black",fill="gray") +
  labs(y="# HR regions") +
    labs(tags = "A") + 
  theme(axis.text = element_text(size = 15),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.y = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
        plot.title = element_text(size = 15,face = "bold")) +
    xlab("Country") +
    coord_flip() 


species_no_lookup <- import_num %>% group_by(Species) %>% summarise(n=n()) %>% mutate(species_ext = paste0(Species," (",n,")"))
import_num <- import_num %>% left_join(species_no_lookup,by = c("Species"))
species_box <- ggplot(subset(import_num,import_num$Species != "STOAT"), aes(x=species_ext, y=X0, fill=Species)) + 
  geom_boxplot(color="black",fill="gray") +
  labs(y="# HR regions") +
    labs(tags = "B") + 
  theme(axis.text = element_text(size = 15),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.y = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
        plot.title = element_text(size = 15,face = "bold")) +
    xlab("Species") +
    coord_flip() 

cluster_no_lookup <- import_num %>% group_by(fastbaps) %>% summarise(n=n()) %>% mutate(cluster_ext = paste0(fastbaps," (",n,")"))
import_num <- import_num %>% left_join(cluster_no_lookup,by = c("fastbaps"))
cluster_box <- ggplot(import_num, aes(x=cluster_ext, y=X0, fill=factor(fastbaps))) + 
  geom_boxplot(color="black",fill="gray") +
  labs(y="# HR regions") + 
  labs(tags = "C") + 
  theme(axis.text = element_text(size = 15),
        axis.text.y = element_text(size = 14),
        axis.text.x = element_text(size = 14),
        axis.title.y = element_text(size = 15),
        axis.title.x = element_text(size = 15),
        panel.background = element_rect(fill = "white"),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        legend.position = "none",
        plot.title = element_text(size = 15,face = "bold")) +
    xlab("Clusters") +
    coord_flip() 
grid.arrange(country_box,species_box,cluster_box,nrow = 2)


```

## Manhattan Plots for Selective Sweep Regions
```{r}
setwd("../dependencies/selective_sweep/")

op_US <- read.table("OmegaPlus_Report.US_mp")
op_US$class <- "USA"
op_UK <- read.table("OmegaPlus_Report.UK_mp")
op_UK$class <- "UK"
op_NZ <- read.table("OmegaPlus_Report.NZ_mp")
op_NZ$class <- "NZ"

op_badger <- read.table("OmegaPlus_Report.badger_mp")
op_badger$class <- "badger"
op_deer <- read.table("OmegaPlus_Report.deer_mp")
op_deer$class <- "deer"
op_possum <- read.table("OmegaPlus_Report.possum_mp")
op_possum$class <- "possum"

op_1 <- read.table("OmegaPlus_Report.1_mp")
op_1$class <- "1"
op_2 <- read.table("OmegaPlus_Report.2_mp")
op_2$class <- "2"
op_4 <- read.table("OmegaPlus_Report.4_mp")
op_4$class <- "4"
op_3 <- read.table("OmegaPlus_Report.3_mp")
op_3$class <- "3"
op_5 <- read.table("OmegaPlus_Report.5_mp")
op_5$class <- "5"
op_6 <- read.table("OmegaPlus_Report.6_mp")
op_6$class <- "6"
op_7 <- read.table("OmegaPlus_Report.7_mp")
op_7$class <- "7"
op_8 <- read.table("OmegaPlus_Report.8_mp")
op_8$class <- "8"


op_dat <- rbind(op_US,op_UK,op_NZ,op_badger,op_deer,op_possum,op_1,op_2,op_3,op_4,op_5,op_6,op_7,op_8)
op_dat$coords <- rownames(op_dat)
colnames(op_dat)[1] <- "omega"

#clean
for (group in 1:length(unique(op_dat$class))){

current <- unique(op_dat$class)[group] 

op_gg <- op_dat %>% filter(class == current) %>% ggplot(aes(x=floor(as.numeric(coords)),y=omega)) + geom_point() +
  theme_minimal() + 
  xlab("genomic position") +
  ylab("omega value") + 
  ggtitle(current)
plot(op_gg)

}

op_dat  %>% ggplot(aes(x=floor(as.numeric(coords)),y=omega)) + geom_point() +
  theme_minimal() + 
  xlab("genomic position") +
  ylab("omega value") + facet_wrap(~ class, ncol = 4)

```

## Regions with significant differences in selective sweeps

```{r}
#We are interested in the regions that are most surprising. 
op_dat <- rbind(op_US,op_UK,op_NZ,op_badger,op_deer,op_possum,op_1,op_2,op_3,op_4,op_5,op_6,op_7,op_8)
op_dat$coords <- floor(as.numeric(rownames(op_dat)))
colnames(op_dat)[1] <- "omega"
avg_lookup <- op_dat  %>% group_by(class) %>% summarise(avg=mean(omega),std=sd(omega),number=n()) 
op_dat_extend <- op_dat %>% left_join(avg_lookup,by="class") %>% 
  mutate(zscore = ((omega - avg)/std)) %>% mutate(p_value = pnorm(zscore, mean = avg, sd = std, lower.tail = FALSE)) %>% mutate(pval.adj = p.adjust(p_value, method='BH'))
op_dat_extend$log_p <- (-1*log10(op_dat_extend$p_value))

#bonferroni pvalue correction - combat multiple hypothesis testing.
bonferroni_corrected = 0.05/5000

for (group in 1:length(unique(op_dat$class))){

current <- unique(op_dat$class)[group] 

op_gg <- op_dat_extend %>% filter(class == current) %>% ggplot(aes(x=floor(as.numeric(coords)),y=log_p)) + geom_point() +
  theme_minimal() + 
  xlab("genomic position") +
  ylab("-log10(P)") + 
  geom_hline(yintercept=7.3, linetype="dashed", color = "black") +
  ggtitle(current)
plot(op_gg) }

op_fig <- op_dat_extend %>% filter(class == "2" ) %>% mutate(Color = ifelse(log_p > 6.3, "black", "gray")) %>% ggplot(aes(x=floor(as.numeric(coords)),y=log_p,color=Color)) + geom_point() +
  theme_minimal() + 
  theme(axis.title = element_text(size = 15)) +
  xlab("genomic position") +
  ylab("-log10(P)") + 
  scale_color_identity() + 
  ylim(c(0,130)) +
  xlim(c(3.6e6,4e6))

```

## Random Forest Analysis 

```{r}
#Initialize the RF models
set.seed(1)
recomb_heat_nolab <-recomb_heat[,-which(names(recomb_heat) %in% c("label"))]
colnames(recomb_heat_nolab) <- paste0("V",colnames(recomb_heat_nolab))

recomb_heat_nolab$Species <- factor(recomb_extend$Species)
mbov_rf <- randomForest(Species ~ .,data = recomb_heat_nolab,importance=TRUE, ntree=10000)

species_rf <- as.data.frame(varImpPlot(mbov_rf))
species_rf$class <- "Species"

recomb_heat_nolab <-recomb_heat[,-which(names(recomb_heat) %in% c("label"))]
colnames(recomb_heat_nolab) <- paste0("V",colnames(recomb_heat_nolab))

recomb_heat_nolab$Country <- factor(recomb_extend$Country)
mbov_rf <- randomForest(Country ~ .,data = recomb_heat_nolab,importance=TRUE, ntree=10000)

country_rf <- as.data.frame(varImpPlot(mbov_rf))
country_rf$class <- "Country"

recomb_heat_nolab <-recomb_heat[,-which(names(recomb_heat) %in% c("label"))]
colnames(recomb_heat_nolab) <- paste0("V",colnames(recomb_heat_nolab))

recomb_heat_nolab$Clusters <- factor(recomb_extend$fastbaps)
mbov_rf <- randomForest(Clusters ~ .,data = recomb_heat_nolab,importance=TRUE, ntree=10000)

cluster_rf <- as.data.frame(varImpPlot(mbov_rf))
cluster_rf$class <- "Population Cluster"
mbov_rf.results <- rbind(species_rf,country_rf,cluster_rf)
```




```{r}
#Compute Variable importance for each HR region
for(i in 1:length(unique(mbov_rf.results$class))){
  current = unique(mbov_rf.results$class)[i]
  sub_data <- subset(mbov_rf.results,class == current)
  
  #For accuracy 
  quant251 <- quantile(sub_data$MeanDecreaseAccuracy,0.25)
  quant751 <- quantile(sub_data$MeanDecreaseAccuracy,0.75)
  iqr1 = quant751 - quant251
  threshold1 = iqr1*1.5 + quant751
  
  #For gini
  quant252 <- quantile(sub_data$MeanDecreaseGini,0.25)
  quant752 <- quantile(sub_data$MeanDecreaseGini,0.75)
  iqr2 = quant752 - quant252
  threshold2 = iqr2*1.5 + quant752
  
  sub_data$importance <- ifelse(sub_data$MeanDecreaseAccuracy > threshold1 & sub_data$MeanDecreaseGini > threshold2, "TRUE", "FALSE")
  print(subset(sub_data,importance == "TRUE"))
  
  plot(sub_data %>% ggplot(aes(x=MeanDecreaseAccuracy, y=MeanDecreaseGini,color = importance)) +
  geom_point() +
    theme_minimal()+
    scale_color_manual(values = c("gray","black")) +
    ggtitle(current))
  
}

## figure out how to put these in a grid?

```

```{r}
    #Cluster
sub_data <- subset(mbov_rf.results,class == "Population Cluster")
  
  #For accuracy 
  quant251 <- quantile(sub_data$MeanDecreaseAccuracy,0.25)
  quant751 <- quantile(sub_data$MeanDecreaseAccuracy,0.75)
  iqr1 = quant751 - quant251
  threshold1 = iqr1*1.5 + quant751
  
  #For gini
  quant252 <- quantile(sub_data$MeanDecreaseGini,0.25)
  quant752 <- quantile(sub_data$MeanDecreaseGini,0.75)
  iqr2 = quant752 - quant252
  threshold2 = iqr2*1.5 + quant752
  
  sub_data$importance <- ifelse(sub_data$MeanDecreaseAccuracy > threshold1 & sub_data$MeanDecreaseGini > threshold2, "TRUE", "FALSE")
  
    c <- sub_data %>% ggplot(aes(x=MeanDecreaseAccuracy, y=MeanDecreaseGini,color = importance)) +
  geom_point() +
    theme_minimal()+
    scale_color_manual(values = c("gray","black")) +
    ggtitle("C") + 
       annotate(geom = "segment", x = 43.8, y = 14.791974, xend = 60, yend = 17) +
       annotate(geom = "text", x = 60, y = 18, label = "Mb0403", hjust = "left") +
       annotate(geom = "segment", x = 44.44055, y = 15.404605, xend = 30, yend = 16.5) +
       annotate(geom = "text", x = 30, y = 17.5, label = "rpfA", hjust = "right") +
       annotate(geom = "segment", x = 43.26362, y = 14.88607, xend = 20, yend = 13) +
        annotate(geom = "text", x = 20, y = 13, label = "rpfA", hjust = "right") +
       annotate(geom = "text", x = 48, y = 5.5, label = "rpfA", hjust = "left") +
       annotate(geom = "segment", x = 40, y = 25, xend = 52.50778, yend = 23.588206) +
       annotate(geom = "text", x = 40, y = 26, label = "rpfA", hjust = "right") +
       annotate(geom = "text", x = 55, y = 26, label = "pks12", hjust = "left") +
       annotate(geom = "text", x = 49, y = 11, label = "pks12", hjust = "left") +
       annotate(geom = "text", x = 50, y = 13, label = "Mb2189c", hjust = "left") +
       annotate(geom = "text", x = 40, y = 6.366, label = "bkdc", hjust = "right") +
       annotate(geom = "text", x = 55, y = 23.451064, label = "Mb3510c", hjust = "left") +
       annotate(geom = "text", x = 73, y = 29.072326	, label = "Mb3510c", hjust = "left") +
       annotate(geom = "text", x = 95, y = 34.504183	, label = "Mb3510c", hjust = "left") +
        theme(
    axis.title = element_text(size = 15),
    plot.title = element_text(size = 15, face = "bold")
  ) +
         labs(color = "Passes Threshold") +
       xlim(c(-10,120)) +
    xlab("MDA") + 
     ylab("MDI") 
```

```{r}

#Species
sub_data <- subset(mbov_rf.results,class == "Species")
  
  #For accuracy 
  quant251 <- quantile(sub_data$MeanDecreaseAccuracy,0.25)
  quant751 <- quantile(sub_data$MeanDecreaseAccuracy,0.75)
  iqr1 = quant751 - quant251
  threshold1 = iqr1*1.5 + quant751
  
  #For gini
  quant252 <- quantile(sub_data$MeanDecreaseGini,0.25)
  quant752 <- quantile(sub_data$MeanDecreaseGini,0.75)
  iqr2 = quant752 - quant252
  threshold2 = iqr2*1.5 + quant752
  
  sub_data$importance <- ifelse(sub_data$MeanDecreaseAccuracy > threshold1 & sub_data$MeanDecreaseGini > threshold2, "TRUE", "FALSE")
  
   a <- sub_data %>% ggplot(aes(x=MeanDecreaseAccuracy, y=MeanDecreaseGini,color = importance)) +
  geom_point() +
    theme_minimal()+
    scale_color_manual(values = c("gray","black")) +
    ggtitle("A") + 
     annotate(geom = "text", x = 55, y = 7.426242, label = "Mb3510c", hjust = "left") +
     annotate(geom = "text", x = 73, y = 10.293916, label = "Mb3510c", hjust = "left") + 
     annotate(geom = "segment", x = 40.79471, y = 3.300782, xend = 55, yend = 4.5) +
     annotate(geom = "text", x = 56, y = 4.8, label = "rpfA", hjust = "left") + 
     annotate(geom = "segment", x = 41.3785, y = 3.279978, xend = 50, yend = 2.3) +
     annotate(geom = "text", x = 51, y = 2.1, label = "rpfA", hjust = "left") + 
     annotate(geom = "segment", x = 41.90512, y = 3.424511, xend = 30, yend = 4) +
     annotate(geom = "text", x = 29, y = 4.2, label = "Mb0403", hjust = "right") +
     xlim(c(-27,85)) + 
     theme(
    axis.title = element_text(size = 15),
    plot.title = element_text(size = 15, face = "bold")
  ) +
         labs(color = "Passes Threshold") +
     xlab("MDA") + 
     ylab("MDI")

```

```{r}
#Country
sub_data <- subset(mbov_rf.results,class == "Country")
  
  #For accuracy 
  quant251 <- quantile(sub_data$MeanDecreaseAccuracy,0.25)
  quant751 <- quantile(sub_data$MeanDecreaseAccuracy,0.75)
  iqr1 = quant751 - quant251
  threshold1 = iqr1*1.5 + quant751
  
  #For gini
  quant252 <- quantile(sub_data$MeanDecreaseGini,0.25)
  quant752 <- quantile(sub_data$MeanDecreaseGini,0.75)
  iqr2 = quant752 - quant252
  threshold2 = iqr2*1.5 + quant752
  
  sub_data$importance <- ifelse(sub_data$MeanDecreaseAccuracy > threshold1 & sub_data$MeanDecreaseGini > threshold2, "TRUE", "FALSE")
  
    b <- sub_data %>% ggplot(aes(x=MeanDecreaseAccuracy, y=MeanDecreaseGini,color = importance)) +
  geom_point() +
    theme_minimal()+
    scale_color_manual(values = c("gray","black")) +
    ggtitle("B") + 
      annotate(geom = "segment", x = 42.78615, y = 15.436363, xend = 62.5, yend = 15) +
     annotate(geom = "text", x = 63, y = 15, label = "Mb0403", hjust = "left") +
     annotate(geom = "text", x = 43, y = 14, label = "rpfA", hjust = "left") + 
    annotate(geom = "text", x = 46, y = 18, label = "rpfA", hjust = "left") + 
      annotate(geom = "text", x = 35, y = 2, label = "rpfA", hjust = "left") + 
      annotate(geom = "text", x = 47, y = 21.347377, label = "rpfA", hjust = "left") + 
      annotate(geom = "text", x = 43, y = 23, label = "pks12", hjust = "right")  + 
      annotate(geom = "text", x = 36, y = 7, label = "bkdc", hjust = "left") + 
     annotate(geom = "text", x = 90, y = 43.319627, label = "Mb3510c", hjust = "right") +
      annotate(geom = "text", x = 67, y = 30.559217, label = "Mb3510c", hjust = "right") +
      annotate(geom = "text", x = 44, y = 19.5, label = "Mb3510c", hjust = "right") +
      annotate(geom = "text", x = 32, y = 16, label = "Mb3510c", hjust = "right") +
      annotate(geom = "text", x = 43, y = 3.299762, label = "Mb3510c", hjust = "left") +
      theme(
    axis.title = element_text(size = 15),
    plot.title = element_text(size = 15, face = "bold")
  ) +
         labs(color = "Passes Threshold") +
     xlab("MDA") + 
     ylab("MDI")
```

```{r}
grid.arrange(a,b,c,ncol = 2)
```
